File: mcp/ts/src/tools/list-brains.ts
// SPDX-License-Identifier: Apache-2.0

import { z } from 'zod'
import { jsonContent, type Tool } from './types.js'

const schema = z.object({}).strict()

export const listBrainsTool: Tool<typeof schema> = {
  name: 'memory_list_brains',
  description: 'List all brains the caller has access to.',
  inputSchema: schema,
  async handler(_args, client) {
    const result = await client.listBrains()
    return jsonContent(result)
  },
}

File: sdks/go/cmd/memory-mcp/tools/list_brains.go
// SPDX-License-Identifier: Apache-2.0

package tools

import (
	"context"

	"github.com/google/jsonschema-go/jsonschema"
	"github.com/modelcontextprotocol/go-sdk/mcp"
)

func registerListBrains(server *mcp.Server, client MemoryClient) {
	schema := &jsonschema.Schema{Type: "object", Properties: map[string]*jsonschema.Schema{}}
	mcp.AddTool(server, &mcp.Tool{
		Name:        "memory_list_brains",
		Description: "List all brains the caller has access to.",
		InputSchema: schema,
	}, func(ctx context.Context, req *mcp.CallToolRequest, _ ListBrainsArgs) (*mcp.CallToolResult, any, error) {
		result, err := client.ListBrains(ctx)
		if err != nil {
			return nil, nil, err
		}
		return structuredResult(result)
	})
}
